src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -633,14 +683,30 @@ RelationalDBInterfaceSqliteImp::saveValidatedLedger(
     /* if databases exists, use it */
     if (existsLedger() && existsTransaction())
     {
-        return ripple::saveValidatedLedger(
-            *lgrdb_, *txdb_, app_, ledger, current);
+        if (!ripple::saveValidatedLedger(
+                *lgrdb_, *txdb_, app_, ledger, current))
+            return false;
+    }
+
+    if (auto shardStore = app_.getShardStore())
+    {
+        auto lgrSesh = lgrMetaDb_->checkoutDb();
+        auto txSesh = txMetaDb_->checkoutDb();
+
+        if (ledger->info().seq < shardStore->earliestLedgerSeq())
+            return true;  // Should be set to false in the future
```

Comment:
What is preventing this from returning false today?

Reply:
Right now the main use for this is storing to the NodeStore SQL databases, so I don't want to return an error simply because the ShardStore doesn't accept a ledger based on its sequence. For the moment I think it makes more sense to return false only if the ShardStore should accept the ledger, but we fail when attempting to save it. In the future when its just the shard database, then I think it would be appropriate to return false if the ledger is too early.

Reply:
@scottschurr Done